mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -652,6 +654,12 @@ public GridType overrideType(Type type) {
 		else if ( type == StandardBasicTypes.BYTE ) {
 			return ByteStringType.INSTANCE;
 		}
+		else if ( type == StandardBasicTypes.SHORT ) {
+			return ShortType.INSTANCE;
+		}
+		else if ( type == StandardBasicTypes.FLOAT ) {
+			return FloatType.INSTANCE;
```

Comment:
you haven't created a specific type for MongoDB, so there is no need to override the type in this case (same for short type)